src/schema/users.ts

```diff
@@ -967,6 +1025,55 @@ export const resolvers: IResolvers<any, Context> = {
         )
         .andWhere('"userId" = :userId', { userId: ctx.userId })
         .execute(),
+    subscribePersonalizedDigest: async (
+      _,
+      args: {
+        hour: number;
+        day: number;
+        timezone: string;
+      },
+      ctx: Context,
+    ): Promise<GQLPersonalizedDigest> => {
+      const { hour, day, timezone } = args;
+
+      if (hour < 0 || hour > 23) {
+        throw new ValidationError('Invalid hour');
+      }
+
+      if (day < 0 || day > 6) {
+        throw new ValidationError('Invalid day');
+      }
+
+      if (Number.isNaN(getTimezoneOffset(timezone))) {
+        throw new ValidationError('Invalid timezone');
+      }
+
+      const repo = ctx.con.getRepository(UserPersonalizedDigest);
+
+      const personalizedDigest = await repo.save({
+        userId: ctx.userId,
+        preferredDay: day,
+        preferredHour: hour,
+        preferredTimezone: timezone,
+      });
+
+      return personalizedDigest;
+    },
+    unsubscribePersonalizedDigest: async (
+      _,
+      args,
+      ctx: Context,
+    ): Promise<unknown> => {
+      const repo = ctx.con.getRepository(UserPersonalizedDigest);
+
+      if (ctx.userId) {
+        await repo.delete({
+          userId: ctx.userId,
+        });
+      }
```

Comment:
Not really comment about your code.
But let's say we roll out now to these 10K users.

And 100 will unsubscribe -> deleting this entry.
Then we decide to enroll everyone by running the script.
These will would be added again correct?

How can we ensure that doesn't happen?

Reply:
Yes, good catch. Since experiment was not meant or planned we can't in current db schema. What we could do is if we decide to enroll everyone check the db before we do it and do not re-add any users who unsubscribed deleted their entry.

Alternative would be to add status `subscribed | unsubscribed` column.

Reply:
Yeah let's opt for the "simplist" one.
Maybe it's enough to simply keep track of these 10K people seperate? (as in CSV) and manually exclude on next step?

Reply:
🤝

Reply:
So my plan is:
- enroll first 10k at random with query in description
- pull them out into CSV file
- when we decide we want to import more users we are gonna manually see which of them unsubscribe to not enroll again